scipy/stats/morestats.py

```diff
@@ -1045,7 +1061,7 @@ def boxcox(x, lmbda=None, alpha=None):
         return special.boxcox(x, lmbda)
 
     # If lmbda=None, find the lmbda that maximizes the log-likelihood function.
-    lmax = boxcox_normmax(x, method='mle')
+    lmax = boxcox_normmax(x, method='mle', optimizer=optimizer)
```

Comment:
We don't want to pass `brack` here?

Reply:
The user can't specify it from `boxcox` (it's not a parameter), only using `boxcox_normmax` correctly. Not passing here results in the default `brack` being used.

Reply:
Yes, this is what I mean. Just wanted to make sure with you that we don't want to allow the user to specify `brack` from `boxcox`.

Reply:
I agree, if the user wants to specify `brack`, they can pass it via `optimizer`, similar to `bounds`.

Reply:
OK no problem, then you can just adapt the doc (saying it will raise and not silently go) and add the corresponding test. Thanks!

Reply:
Thanks @tupui - I updated the docs and added the test.